extensions/gdata/src/com/google/refine/extension/gdata/GoogleAPIExtension.java

```diff
@@ -0,0 +1,205 @@
+package com.google.refine.extension.gdata;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.servlet.http.HttpServletRequest;
+
+import com.google.api.client.auth.oauth2.AuthorizationCodeResponseUrl;
+import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeRequestUrl;
+import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeTokenRequest;
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.auth.oauth2.GoogleTokenResponse;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson.JacksonFactory;
+import com.google.api.services.drive.Drive;
+import com.google.api.services.drive.DriveScopes;
+import com.google.api.services.fusiontables.FusiontablesScopes;
+import com.google.api.services.sheets.v4.Sheets;
+import com.google.api.services.sheets.v4.SheetsScopes;
+
+import com.google.refine.util.ParsingUtilities;
+
+import edu.mit.simile.butterfly.ButterflyModule;
+
+abstract public class GoogleAPIExtension {
+    protected static final String SERVICE_APP_NAME = "OpenRefine-Google-Service";
+    private static final String CLIENT_ID = "455686949425-d237cmorii0ge8if7it5r1qijce6caf0.apps.googleusercontent.com";
```

Comment:
@jackyq2015 hmm, where how are these tokens maintained ? oh wait, are these even tokens , or registered ids with Google Apps somewhere ? I'm just wondering if Tom maintained an account for this stuff or not , and if we should know about this more from him ?

Reply:
It is a registered ids but with new set of APIs enabled. I created the Id

Reply:
Is this tied to your own Google account somehow? And is this token really meant to be public? (Could other applications maliciously reuse this token?) (Not that I can see any way to avoid that…)